sway_libs/src/merkle_proof.sw

```diff
@@ -0,0 +1,151 @@
+// TODO: Function definitions that use arrays should be updated to Vecs once
+// https://github.com/FuelLabs/fuels-rs/issues/353 is resolved
+
+library merkle_proof;
+
+use std::{hash::sha256, option::Option, revert::{require, revert}, vec::Vec};
+
+pub enum ProofError {
+    InvalidMultiProof: (),
+}
+
+fn hash_pair(a: b256, b: b256) -> b256 {
+    if a <= b {
+        // Hash(a + b)
+        sha256((a, b))
+    } else {
+        // Hash(b + a)
+        sha256((b, a))
+    }
+}
```

Comment:
This looks like you're trying to copy Uniswap's Merkle tree. This library should instead conform to https://github.com/FuelLabs/fuel-merkle-sol, which implements [RFC-6962](https://datatracker.ietf.org/doc/html/rfc6962) Merkle tree proof verification.